src/api/FeelingsAPI/index.ts

```diff
@@ -29,22 +37,43 @@ export const getFeelingsOnDate = async (date: Date) => {
   return feelingsList;
 };
 
-export const getFeelingsBetweenDates = async (startDate : Date, endDate : Date) => {
+export const getFeelingsBetweenDates = async (startDate: Date, endDate: Date) => {
   db.transaction('rw', db.feelingsCollection, async () => {
     const feelingsList = await db.feelingsCollection.where('date').between(startDate, endDate);
     return feelingsList;
   });
 };
 
-export const addFeeling = async (feelingName : string, feelingCategory : string) => {
-  const currentDate = getJustDate(new Date());
-  const currentDateFeelings = await getFeelingsOnDate(currentDate);
-  const checkFeelings = (feeling:FeelingItem) => feeling.content === feelingName;
+export const addFeeling = async (feelingName: string, feelingCategory: string, feelingDate: Date) => {
+  // const currentDate = getJustDate(new Date());
+  const feelingDateFormatted = getJustDate(feelingDate);
+  const currentDateFeelings = await getFeelingsOnDate(feelingDate);
+  const checkFeelings = (feeling: IFeelingItem) => feeling.content === feelingName;
   if (currentDateFeelings.some(checkFeelings)) { return; }
   db.transaction('rw', db.feelingsCollection, async () => {
     await db
       .feelingsCollection
-      .add({ content: feelingName, date: currentDate, category: feelingCategory });
+      .add({ content: feelingName, date: feelingDateFormatted, category: feelingCategory });
+  }).catch((e) => {
+    console.log(e.stack || e);
+  });
+};
+
+export const addFeelingNote = async(feelingId: number, InputNote: string) => {
+  const feeling = await db.feelingsCollection.get(feelingId);
+  const updatedFeeling = {...feeling, note: InputNote};
+  db.transaction('rw', db.feelingsCollection, async () => {
+    await db.feelingsCollection.put(updatedFeeling);
+  }).catch((e) => {
+    console.log(e.stack || e);
+  });
```

Comment:
Right after this line make a call get the updated feelings and return

Reply:
Exactly what I did